moveit_planners/chomp/chomp_interface/CMakeLists.txt

```diff
@@ -63,7 +63,7 @@ if(CATKIN_ENABLE_TESTING)
     test/chomp_moveit.test
     test/chomp_moveit_test.cpp)
   target_link_libraries(chomp_moveit_test
-    ${moveit_ros_planning_interface_LIBRARIES}
     ${catkin_LIBRARIES}
+    ${moveit_ros_planning_interface_LIBRARIES}
```

Comment:
Is this semantically different?

Reply:
Yes.

- catkin includes libraries of all dependencies of the package in here (that list is 86 elements long on my system)

- the standard GNU toolchain uses the first appearance of a required function symbol and ignores later appearances

As a result you end up linking a library from an overlay (a dependency of moveit_ros_planning_interface), but included the headers of the overlayed package above.

Changing this does not change the problem though that adding /opt/ros/<distro>/include/ in `catkin_INCLUDE_DIRS` will accidentally provide access to the headers of `moveit_ros_planning_interface` that might be overlayed in a higher workspace. In that case the build will still use system headers but link the libraries added here.